Orange/widgets/tests/base.py

```diff
@@ -37,7 +37,23 @@
 from Orange.widgets.widget import OWWidget
 
 
-WidgetTest = WidgetTestBase
+class WidgetTest(WidgetTestBase):
+    def assert_table_equal(self, table1, table2):
+        if table1 is None or table2 is None:
+            self.assertIs(table1, table2)
+            return
+        self.assert_domain_equal(table1.domain, table2.domain)
+        np.testing.assert_array_equal(table1.X, table2.X)
+        np.testing.assert_array_equal(table1.Y, table2.Y)
+        np.testing.assert_array_equal(table1.metas, table2.metas)
+
+    def assert_domain_equal(self, domain1, domain2):
+        for var1, var2 in zip(domain1.variables + domain1.metas,
+                              domain2.variables + domain2.metas):
+            self.assertEqual(type(var1), type(var2))
+            self.assertEqual(var1.name, var2.name)
+            if var1.is_discrete:
+                self.assertEqual(var1.values, var2.values)
```

Comment:
Currently, this is different than `domain1 == domain2`. Maybe we should fix `DiscreteVariable` to check `values` in `__eq__`.
But despite that, do we really want to "reimplement" domain equality here? Is it for debugging purposes (a specific assert fails in case of nonequality)?

Reply:
I added a docstring.